fontbe/src/features.rs

```diff
@@ -67,12 +78,183 @@ impl Display for NotSupportedError {
     }
 }
 
+// TODO: ask Colin if we need to be threadsafe
+struct FeaVariationInfo<'a> {
+    fea_rs_axes: HashMap<Tag, (&'a CoordConverter, fea_rs::compile::AxisInfo)>,
+    axes: HashMap<Tag, &'a Axis>,
+    static_metadata: &'a StaticMetadata,
+    /// fea coords, unlike glyphs, use locations we don't know until quite late.
+    ///
+    /// We have to establish a new, somewhat expensive, model for each unique combination of
+    /// locations we observe. We also use different `ModelConstraints` than glyf: we don't require
+    /// the effect at the default location to be zero.
```

Comment:
> We also use different `ModelConstraints` than glyf: we don't require the effect at the default location to be zero.

sorry but that's wrong. The effect at the default location is always 0 throughout the OT Variations, it's not about glyf or GPOS. The variable scalars in FEA must always specify a value for the default master's location, there's no exception here.